logistics/scripts/modules/settings/settings-management.psm1

```diff
@@ -0,0 +1,351 @@
+    return @{
+        Ods      = 'EdFi_Ods'
+        Admin    = 'EdFi_Admin'
+        Security = 'EdFi_Security'
+        Master   = 'EdFi_Master'
+    }
+}
+
+function Get-ConnectionStringsByEngine {
+    return  @{
+        SQLServer  = @{
+            ConnectionStrings = @{
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Ods'])      = "Server=(local); Trusted_Connection=True; Database=EdFi_{0};"
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Admin'])    = "Server=(local); Trusted_Connection=True; Database=EdFi_Admin;"
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Security']) = "Server=(local); Trusted_Connection=True; Database=EdFi_Security; Persist Security Info=True;"
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Master'])   = "Server=(local); Trusted_Connection=True; Database=master;"
+            }
+            DatabaseTemplate  = @{
+                MinimalTemplateScript   = 'EdFiMinimalTemplate'
+                PopulatedTemplateScript = 'GrandBend'
+            }
+        }
+        PostgreSQL = @{
+            ConnectionStrings = @{
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Ods'])      = "Host=localhost; Port=5432; Username=postgres; Database=EdFi_{0};"
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Admin'])    = "Host=localhost; Port=5432; Username=postgres; Database=EdFi_Admin;"
+                ((Get-ConnectionStringKeyByDatabaseTypes)['Security']) = "Host=localhost; Port=5432; Username=postgres; Database=EdFi_Security;"
+            }
+            DatabaseTemplate  = @{
+                MinimalTemplateScript   = 'PostgreSQLMinimalTemplate'
+                PopulatedTemplateScript = 'PostgreSQLPopulatedTemplate'
+            }
+        }
+    }
+}
+
+function Get-SubtypesByFeature {
+    return @{
+        changeQueries               = 'changes'
+        ownershipBasedAuthorization = 'RecordOwnership'
+    }
+}
+
+function Test-AppSettings {
+    param(
+        [Parameter(ValueFromPipeline = $true)]
+        [string] $Path
+    )
+
+    if (-not (Test-Path $Path)) { return $false }
+
+    try {
+        Get-Content $Path | ConvertFrom-Json | Out-Null
+        return $true
+    }
+    catch {
+        return $false
+    }
+}
+
+function Assert-ValidAppSettings([string[]] $SettingsFiles = (Get-ChildItem "$(Get-RepositoryRoot 'Ed-Fi-ODS-Implementation')\**\appsettings*.json" -Recurse)) {
+    $result = @()
+
+    foreach ($file in $SettingsFiles) {
+        if (Test-AppSettings $file) {
+            $result += @{ file = $file; success = $true }
+        }
+        else {
+            try {
+                Get-Content $file | ConvertFrom-Json
+            }
+            catch {
+                $result += @{ file = $file; success = $false; exception = $_ }
+            }
+        }
+    }
+
+    return $result
+}
+
+function Add-ApplicationNameToConnectionStrings([hashtable] $ConnectionStrings = @{ }, [string] $ApplicationName) {
```

Comment:
This function seems to be assuming a lot.  It assumes that the connection string is already set up to end with a semicolon (optional), and only changes the Application Name if its not already there.  Does it make more sense to do these operations against the ConnectionStringBuilder instead?  Then you can update (or continue to ignore it) if its there, and you can also not worry about existing connection string formatting.

Reply:
modified this to use the connection string builders